File: 053/Maximum_Subarray.c
#define MAX_INT ((unsigned)(-1)>>1)
#define MIN_INT (~MAX_INT)
int maxSubArray(int* nums, int numsSize) {
    int ret = MIN_INT, tmpSum = 0, i;

    for (i = 0; i < numsSize; i++) {
        tmpSum = tmpSum <= 0 ? nums[i] : (tmpSum + nums[i]);

        if (tmpSum > ret) {
            ret = tmpSum;
        }
    }
    return ret;
}

File: 053/Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = float("-inf")
        tmp_sum = 0

        for x in nums:
            if tmp_sum < 0:
                tmp_sum = x
            else:
                tmp_sum += x

            if tmp_sum > res:
                res = tmp_sum

        return res
